frontend/coprs_frontend/coprs/logic/builds_logic.py

```diff
@@ -798,11 +798,19 @@ def create_new(cls, user, copr, source_type, source_json, chroot_names=None, pkg
         return build
 
     @classmethod
-    def _setup_batch(cls, batch, after_build_id, with_build_id, user):
-        # those three are exclusive!
-        if sum([bool(x) for x in
-                [batch, with_build_id, after_build_id]]) > 1:
-            raise BadRequest("Multiple build batch specifiers")
+    def setup_batch(cls, after_build_id, with_build_id, user,
+                    always_create=False):
+        """
+        According to "after_build_id" and "with_build_id" arguments (they are
+        mutually exclusive), get a pre-existing (or create a new) batch and
+        return it.  If neither of those args is specified, None is returned,
+        unless always_create=True - then a completely separate batch is crated
+        and returned.
+        """
+        batch = None
+        if bool(with_build_id) and bool(after_build_id):
```

Comment:
Are the bools here necessary? They are just numbers or None so the if statement should work either way

e.g.:

```
>>> bool(1 and None)
False
>>> bool(1 and 1)
True
>>> bool(None and None)
False
>>>
```

Reply:
updated